### ISSUE TITLE
[2024-11-07] Add a button to create a contact in the POI form.

### ISSUE THREAD

**Issue by MizukiTemma** [digitalfabrik/integreat-cms#3088]
### Motivation

A new feature [contact](https://github.com/digitalfabrik/integreat-cms/issues/2955) is being implemented. It helps users if there is a possibility to create a contact directly in the POI form.

### Proposed Solution

Implement an ajax form like POI creation in the event form.

### Alternatives

Don't implement.

**Comment by PeterNerlich** [comment]
> Is the relationship between POI/Location and Contact a 1:0..1 relationship?

No, Location and Contact are in a 0..1:0..n relationship: A Contact MAY reference a Location, there can be Locations that are not referenced by any Contact. But there is an additional constraint on the model that there can be at most one Contact with an empty title / *point of contact for* field per referenced Location

**Comment by jarlhengstmengel** [comment]
> So a contact can be created without referencing a location?

NVM

**Comment by PeterNerlich** [comment]
No, that's my mistake, I think a contact always has to reference a location after all

**Comment by jarlhengstmengel** [comment]
@osmers the question came up if the creation of a contact in the POI form should only be available during the editing of  a POI or also during the creation?

**Comment by osmers** [comment]
@jarlhengstmengel how are we handling that with events? I think I would implement the same process here.

**Comment by jarlhengstmengel** [comment]
> @jarlhengstmengel how are we handling that with events? I think I would implement the same process here.

Events is doing both I think. Contact creation during POI creation is a bit more complicated as the creation during editing since a contact needs to reference a POI, but it should be doable. I will probably split this issue into two PRs then

**Comment by osmers** [comment]
@jarlhengstmengel that's a fair point though - I hadn't thought of this...
We could do it this way:
1st Prio - contact creation once POI is already saved
2nd Prio - contact creation before POI is saved (not really that urgent, I think, especially if it would delay the release. I think it's always an improvement we can release later)
What do you think?

**Comment by jarlhengstmengel** [comment]
Sounds good to me :)
